digdag-core/src/main/java/io/digdag/core/storage/ArchiveManager.java

```diff
@@ -99,9 +114,17 @@ private String formatFilePath(int siteId, String projectName, String revisionNam
     {
         Preconditions.checkArgument(!Strings.isNullOrEmpty(projectName), "projectName");
         Preconditions.checkArgument(!Strings.isNullOrEmpty(revisionName), "revisionName");
+        // It's possible the storage used to store a project archive in doesn't accept some characters.
+        // For instance, AWS S3 says the following characters are generally safe
+        //   - Alphanumeric characters [0-9a-zA-Z]
+        //   - Special characters !, -, _, ., *, ', (, and )
+        //
+        // So we'd better encode a project name into the characters above
+        String encodedProjectName = Base64.getEncoder().encodeToString(projectName.getBytes(UTF_8)).replace("=", "_");
```

Comment:
I think what we want to use here is base64url without padding (defined in RFC 4648).
General base64 uses `/` character. `=` in base64 is padding which is OK to remove.
You can build this variant of Base64 by using Guava:

```java
import com.google.common.io.BaseEncoding;
BaseEncoding.base64Url().omitPadding().encode(projectName.getBytes(UTF_8));
```